hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -1089,13 +1089,7 @@ public DatanodeInfo[] getDatanodeReport(DatanodeReportType type)
   DatanodeInfo[] getCachedDatanodeReport(DatanodeReportType type)
       throws IOException {
     try {
-      DatanodeInfo[] dns = this.dnCache.get(type);
-      if (dns == null) {
-        LOG.debug("Get null DN report from cache");
-        dns = getCachedDatanodeReportImpl(type);
-        this.dnCache.put(type, dns);
-      }
-      return dns;
+      return this.dnCache.get(type);
```

Comment:
Thank you for your contribution! but I am not sure that this modification is correct. I need other partners to help review.

@ayushtkn @goiri Can you help review this pr? Thank you very much!

Reply:
I am not sure if it is correct, atleast I am not sure & it makes my code look unsafe

Reply:
@slfan1989 @ayushtkn Thanks for your review, this modification is just to test whether spotbug will pass. Why do I still report spotbugs exception after deleting the relevant code?